libraries/ESP8266SSDP/ESP8266SSDP.cpp

```diff
@@ -150,7 +150,30 @@ SSDPClass::SSDPClass() :
 }
 
 SSDPClass::~SSDPClass() {
-  delete _timer;
+#ifdef DEBUG_SSDP
```

Comment:
1. Instead of putting this directly in the destructor, I suggest implementing the cleanup in a standalone method, e.g.: close(), then calling that method from the destructor. To be specific, the cleanup done in the destructor should reflect the resource acquisition in the constructors + the call to close(), while the cleanup in close() should reflect the acquisition done in begin().
2. close() should be called at the start of begin() to allow calling begin() repeatedly.
3. There is some duplication between the implemented cleanup and the begin() method body. That should be cleaned up.
4. strings should be moved to flash, even (especially?) DEBUG ones.